src/settings/target/template_config.rs

```diff
@@ -0,0 +1,112 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::terminal::message;
+
+use super::{Environment, KvNamespace, Site, TargetType};
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct TemplateConfig {
+    pub account_id: Option<String>,
+    pub env: Option<HashMap<String, Environment>>,
+    #[serde(rename = "kv-namespaces")]
+    pub kv_namespaces: Option<Vec<KvNamespace>>,
+    #[serde(rename = "type")]
+    pub target_type: TargetType,
+    pub route: Option<String>,
+    pub routes: Option<HashMap<String, String>>,
+    pub webpack_config: Option<String>,
+    pub workers_dev: Option<bool>,
+    pub zone_id: Option<String>,
+    pub site: Option<Site>,
+}
+
+impl TemplateConfig {
+    pub fn warn_on_account_info(&self) {
+        let mut top_level_fields: Vec<String> = Vec::new();
+        if let Some(account_id) = &self.account_id {
```

Comment:
If &self.account_id is none, shouldn't we also add this to the list of missing top level fields?

Reply:
ideally this field won't even be an option though (let's re-use Manifest).

Reply:
we want to be able to run generate on invalid tomls which is why everything is an option

Reply:
previously wrangler generate worked without a toml altogether, so we want to be able to support those cases

Reply:
^ this is why i suggest using serde tricks to handle empty fields without using options :)